### PR TITLE
gh-128342: Specify timeout unit in subprocess docstrings

### PR DESCRIPTION
**Pull request by n-l-i**
* Issue: gh-128342

### CODE REVIEW

**Comment by srinivasreddy** [comment]
isn't the implicit assumption for `timeout` is seconds?

**Comment by n-l-i** [comment]
It is not uncommon for `timeout` to be specified in milliseconds in other frameworks and programming languages. I am not aware of whether seconds is the norm unless otherwise specified in the cpython standard libraries, and I haven't used these libraries exclusively enough to have been able to form that impression on my own. That's why I thought it was better to make it clear like has already been done in some other modules here.

File: Lib/subprocess.py [hunk 1]
```diff
@@ -386,7 +386,8 @@ def _text_encoding():
 
 def call(*popenargs, timeout=None, **kwargs):
     """Run command with arguments.  Wait for command to complete or
-    timeout, then return the returncode attribute.
+    for timeout, specified in seconds, to expire. Then return the
+    returncode attribute.
```
Comment (terryjreedy) [inline Lib/subprocess.py]: I think this simpler edit is sufficient.
```suggestion
    for timeout seconds, then return the returncode attribute.
```

File: Lib/subprocess.py [hunk 2]
```diff
@@ -523,8 +524,8 @@ def run(*popenargs,
     in the returncode attribute, and output & stderr attributes if those streams
     were captured.
 
-    If timeout is given, and the process takes too long, a TimeoutExpired
-    exception will be raised.
+    If timeout, specified in seconds, is given and the process takes too
+    long, a TimeoutExpired exception will be raised.
```
Comment (terryjreedy) [inline Lib/subprocess.py]: Ditto.
```suggestion
    If timeout (seconds) is given and the process takes too long,
     a TimeoutExpired exception will be raised.
```

**Review by terryjreedy — CHANGES_REQUESTED** [review]

**Comment by terryjreedy** [comment]
I would merge this with the simpler edits that leave a smoother flow of thought.  Agree with skipping news.

**Review by gpshead — APPROVED** [review]
in python, yes, floating point seconds is the common thing. but API design wise it is _always_ best to explicitly document that.  (given a time machine i'd have even put the units in the parameter name)

**Pull request merged**